lib/instana/instrumentation/rack.rb

```diff
@@ -76,6 +76,11 @@ def call(env)
           ::Instana.tracer.log_error(nil)
         end
 
+        # If the framework instrumentation provides a path template,
+        # pass it into the span here.
+        # See: https://www.instana.com/docs/tracing/custom-best-practices/#path-templates-visual-grouping-of-http-endpoints
+        kvs[:http][:path_tpl] = env['INSTANA_HTTP_PATH_TEMPLATE'] if env['INSTANA_HTTP_PATH_TEMPLATE']
```

Comment:
Why an env var? We should collect all path templates, regardless of how they look like

Reply:
Only one of our supported frameworks ([Sinatra](https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1013)) exposes the pattern which was matched during the request lifecycle. For the rest of the supported frameworks we need to create framework specific plugins (like `CubaPathTemplateExtractor`) to recreate this information. Because of the need for custom instrumentation for each framework, there will be cases where the Rack instrumentation will run without being able to get access to the path template.  Using Rack's env hash felt like the cleanest way to pass the template out to the root span, while letting Rack based frameworks that don't have custom instrumentation still run with the latest version of `ruby-sensor`.

If we don't want to use Rack's env hash, we could update `CubaPathTemplateExtractor#call` to traverse the current span's ancestors till it finds the root span then set `http.path_tpl`.

Reply:
What's going to happen if we're handling multiple requests concurrently, i.e. with Thin or any other app servers? Wouldn't `env` be shared between Is it possible that request B catches the value set by A? Can we store this data in the request for example?

Reply:
`env` is a per request object generated by the server conforming to the [Rack interface](https://github.com/rack/rack/blob/master/SPEC.rdoc#the-environment-). For example, [Thin](https://github.com/macournoyer/thin/blob/ff003f09e4fc338d5c7d4f6834257328a8f6aed5/lib/thin/connection.rb#L30) creates a new `env` object after each thread has finished processing a request.